lib/bank_accounts/account.rb

```diff
@@ -0,0 +1,125 @@
+# throw error it's more of a prog thing like a data type that won't be recognized by a method.
+# if it's more the user trying to take too much, return false b/c the method call is where the error message should trigger.
+
+# be consistent about throwing errors vs false.
+
+# PRIVATE might be the cause / PROTECTED might be the solution
+# nope not returning true/false
+# !R private vs protected
+
+
+module BankAccounts
+  class Account
+    ##--------------------------------------------------------------------------
+    # CONSTANTS
+
+
+    # limits
+    MINIMUM_BALANCE = 0
+
+
+    ##--------------------------------------------------------------------------
+    # INSTANCE METHODS
+
+
+    attr_reader :balance, :id
+
+
+    def initialize(id, initial_balance)
+      if initial_balance >= MINIMUM_BALANCE
+        @balance = 0
+        update_balance(initial_balance.round(2))
+      else
+        raise ArgumentError.new("An account cannot be created with a negative balance.")
```

Comment:
You could also do this like so: 
```
def initialize(id, initial_balance)
  raise ArgumentError.new("An account cannot ... ") if initial_balance < MINIMUM_BALANCE
  @balance = 0
  update_balance(initial_balance.round(2))
```
Since `raise` causes the method to exit, the rest of the method will not be executed if the error is raised, so you can get rid of the whole if/else block. This pattern is called a guard clause. Of course the way you did it is totally fine (and does exactly the same thing) but this lets you cut out some text.